File: ai/app/models/hospital_chat.py
from typing import Any

from pydantic import BaseModel


class StartSessionRequest(BaseModel):
    # 세션 시작은 별도 입력값이 없어서 빈 바디만 받는다.
    pass


class ChatMessageRequest(BaseModel):
    message: str


class ChatData(BaseModel):
    # 프론트가 대화 결과를 후속 행동으로 연결할 수 있도록 최소 메타데이터를 함께 보낸다.
    intent: str
    action_required: str
    hospital_id: int | None = None
    reservation_id: int | None = None
    requested_at: str | None = None
    confirmed_at: str | None = None
    suggested_slots: list[str] | None = None
    reservation_status: str | None = None
    items: list[dict[str, Any]] | None = None


class ChatPayload(BaseModel):
    # 공통 응답의 data 영역에 실제 병원 상담 결과를 담는다.
    conversation_id: str
    message: str
    data: ChatData | dict[str, Any] | None

File: frontend/src/api/endpoints/hospitalChat.ts
import axios from 'axios'

type ApiEnvelope<T> = {
  success: boolean
  message: string
  data: T | null
  error: {
    code: string
    detail: string | null
  } | null
}

export type HospitalChatItem = Record<string, unknown>

export type HospitalChatData = {
  intent: string
  action_required: string
  hospital_id: number | null
  reservation_id: number | null
  requested_at: string | null
  confirmed_at: string | null
  suggested_slots: string[] | null
  reservation_status: string | null
  items: HospitalChatItem[] | null
}

export type HospitalChatPayload = {
  conversation_id: string
  message: string
  data: HospitalChatData | Record<string, unknown> | null
}

const defaultHospitalChatBaseUrl = import.meta.env.DEV
  ? '/ai-api'
  : import.meta.env.VITE_API_BASE_URL || 'https://api.nova-bank.site'
const hospitalChatBaseUrl =
  import.meta.env.VITE_AI_API_BASE_URL || defaultHospitalChatBaseUrl

const hospitalChatClient = axios.create({
  baseURL: hospitalChatBaseUrl,
  timeout: 10000,
  withCredentials: true,
  headers: {
    'Content-Type': 'application/json',
  },
})

export const hospitalChatApi = {
  startSession: async (): Promise<HospitalChatPayload> => {
    const response = await hospitalChatClient.post<ApiEnvelope<HospitalChatPayload>>('/chat', {})
    return response.data.data as HospitalChatPayload
  },

  sendMessage: async (
    conversationId: string,
    message: string
  ): Promise<HospitalChatPayload> => {
    const response = await hospitalChatClient.post<ApiEnvelope<HospitalChatPayload>>(
      `/chat/${conversationId}`,
      { message }
    )
    return response.data.data as HospitalChatPayload
  },

  endSession: async (conversationId: string): Promise<HospitalChatPayload> => {
    const response = await hospitalChatClient.delete<ApiEnvelope<HospitalChatPayload>>(
      `/chat/${conversationId}`
    )
    return response.data.data as HospitalChatPayload
  },
}
